esi_leap/api/controllers/v1/lease.py

```diff
@@ -114,9 +115,13 @@ def get_all(self, project_id=None, start_time=None, end_time=None,
         lease_collection = LeaseCollection()
         leases = lease_obj.Lease.get_all(filters, request)
         lease_collection.leases = []
-        for lease in leases:
-            lease_collection.leases.append(
-                Lease(**LeasesController._lease_get_dict_with_names(lease)))
+        if len(leases) > 0:
+            project_list = keystone.get_project_list()
+            node_list = ironic.get_node_list()
```

Comment:
Does context parameter need to be passed to get_node_list()?

Reply:
Nope! The context parameter is the user context, which would limit the retrieved nodes to the one that the user can see in Ironic. But if we want them to see the names of nodes that are on offer, then we use no context (which results in an ironic client that uses the admin context), so that they can see all of those nodes.

Reply:
oh, I see! Thanks!